cmd/revad/svcs/grpcsvcs/preferencessvc/preferencessvc.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2018-2019 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package preferencessvc
+
+import (
+	"context"
+	"io"
+
+	rpcpb "github.com/cs3org/go-cs3apis/cs3/rpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	preferencesv0alphapb "github.com/cs3org/go-cs3apis/cs3/preferences/v0alpha"
+	"github.com/cs3org/reva/cmd/revad/grpcserver"
+)
+
+func init() {
+	grpcserver.Register("preferencessvc", New)
+}
+
+// m maps user to map of user preferences.
+// m = map[userToken]map[key]value
+var m = make(map[string]map[string]string)
+
+type service struct{}
+
+// New returns a new PreferencesServiceServer
+func New(m map[string]interface{}, ss *grpc.Server) (io.Closer, error) {
+	service := &service{}
+	preferencesv0alphapb.RegisterPreferencesServiceServer(ss, service)
+	return service, nil
+}
+
+func (s *service) Close() error {
+	return nil
+}
+
+func (s *service) SetKey(ctx context.Context, req *preferencesv0alphapb.SetKeyRequest) (*preferencesv0alphapb.SetKeyResponse, error) {
+	//	log := appctx.GetLogger(ctx)
+	key := req.Key
+	value := req.Val
+
+	md, ok := metadata.FromIncomingContext(ctx)
```

Comment:
You don't need to use this. You need to extract the user from the context, you can obtain it like we do it here:

https://github.com/cs3org/reva/blob/master/pkg/storage/fs/eos/eos.go#L108

That is more elegant, change your code accordingly.

Reply:
Done!